supervision/detection/base_annotate.py

```diff
@@ -0,0 +1,37 @@
+from typing import Union
+
+from supervision.draw.color import Color, ColorPalette
+
+
+class BaseAnnotator:
```

Comment:
I like the `BaseAnnotator` idea, but we should drop all text-related properties from the class definition. Not every annotator will draw text. 

I'm not even sure we need `color` and `thickness` is the right choice. I was thinking about making `BlurAnnotator`. As the name suggests, it is responsible for blurring detections. If  you think about it, it doesn't need `color` nor `thickness`.